Simplified/NYPLBarcode.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+import ZXingObjC
+
+fileprivate let barcodeHeight: CGFloat = 100
+fileprivate let maxBarcodeWidth: CGFloat = 500
+
+@objc enum NYPLBarcodeType: Int {
+  case codabar
+  case code39
+  case qrCode
+  case code128
+}
+
+fileprivate func ZXBarcodeFormatFor(_ NYPLBarcodeType:NYPLBarcodeType) -> ZXBarcodeFormat {
+  switch NYPLBarcodeType {
+  case .codabar:
+    return kBarcodeFormatCodabar
+  case .code39:
+    return kBarcodeFormatCode39
+  case .qrCode:
+    return kBarcodeFormatQRCode
+  case .code128:
+    return kBarcodeFormatCode128
+  }
+}
+
+/// Manage creation and scanning of barcodes on library cards.
+/// Keep any third party dependency abstracted out of the main app.
+final class NYPLBarcode: NSObject {
+
+  class func image(fromString string: String, superviewWidth: CGFloat, type: NYPLBarcodeType) -> UIImage?
```

Comment:
On the move from Swift 2 to Swift 3, Apple completely renamed every method in their Foundation and UI API's from the extremely verbose, almost sentence-like structure to something that takes out all unneccessary vocabulary that either does not describe the method, or sounds redundent between the method and parameter names.

I agree that "image(fromString)..." can seem vague until you look at the callsite and read it fully with the class name prepended:
`let variableName = NYPLBarcode.image(fromString: "string" ....)`
get and set are also implicit on usage, as has been for some time on iOS, so those verbs are also ommitted.

On the second point, since `-> UIImage?` denotes that you either get back a Drawable, or null, receiving "null" is the failure case and is an extremely common pattern for quick and simple methods like this. More complex API's can implement the `throws` keyword and provide an error type enum to enforce the caller to handle all error cases.

Swift uses a documenting structure based on Markdown syntax. There are no header files, so you'd have to place above every public method you're describing. I personally felt there was enough context in the class description to fill in the two methods, but if @winniequinn thinks there needs to be improvement I'll accept the majority vote.